crates/ctranslate2-bindings/build.rs

```diff
@@ -37,9 +52,21 @@ fn link_static() -> PathBuf {
     if cfg!(target_os = "linux") {
         config
             .define("WITH_MKL", "OFF")
-            .define("OPENMP_RUNTIME", "NONE")
-            .define("WITH_OPENBLAS", "ON")
-            .cxxflag("-msse4.1")
+            .define("OPENMP_RUNTIME", "NONE");
+
+        if cfg!(target_arch = "aarch64") && check_jetson() {
+            config
+               .define("WITH_CUDA", "ON")
+               .define("WITH_CUDNN", "ON")
+               .cxxflag("-mcpu=native")
+        } else if cfg!(target_feature = "sse4.1") {
+            config
+                .define("WITH_OPENBLAS", "ON")
+                .cxxflag("-msse4.1")
+        } else {
+            config
+                .define("WITH_CUDA", "ON")
+        }
```

Comment:
Could you refactor above block with following branching structures?

```rust
if cfg!(target_os = "linux") {
  ...
  if cfg!(target_feature = "sse4.1") {
    ...
  }

  if cfg!(feature = "link_static_cuda") {
    // turn on cuda
    ...
    // if needed, branch here on aarch64 / x86
  } else {
    // Turn on openblas
    ...
  }
} else if cfg!(target_os = "macos") {
  ...
}
```

Reply:
Copy that! I will try to refactor it ASAP

Reply:
@wsxiaoys I've refactored as you requested. Please take a look at this revision